src/batch/src/executor/join/lookup_join.rs

```diff
@@ -317,263 +316,191 @@ impl<P: 'static + ProbeSideSourceBuilder> Executor for LookupJoinExecutor<P> {
-            let mut probe_side_stream = probe_child.execute();
-
-            let mut probe_side_chunk_exists = true;
-
-            // for each row in the build side chunk, has_match tells us if it has a match
-            // on the probe side table
-            let mut has_match = all_row_refs
-                .iter()
-                .map(|rows| vec![false; rows.len()])
-                .collect_vec();
+            // Build hash map
+            for (build_chunk_id, build_chunk) in build_side.iter().enumerate() {
+                let build_keys = K::build(&hash_join_build_side_key_idxs, build_chunk)?;
 
-            // Keep looping until there are no more probe side chunks to get, then do 1 more loop
-            while probe_side_chunk_exists {
-                let probe_side_chunk = probe_side_stream.next().await;
-                probe_side_chunk_exists = probe_side_chunk.is_some();
+                for (build_row_id, build_key) in build_keys.into_iter().enumerate() {
+                    // Only insert key to hash map if it is consistent with the null safe
+                    // restriction.
+                    if build_key.null_bitmap().is_subset(&null_matched) {
+                        let row_id = RowId::new(build_chunk_id, build_row_id);
+                        next_build_row_with_same_key[row_id] = hash_map.insert(build_key, row_id);
+                    }
+                }
+            }
 
-                let probe_side_chunk = if let Some(chunk) = probe_side_chunk {
-                    Some(chunk?.compact()?)
-                } else {
-                    None
+            let params = EquiJoinParams::new(
+                hash_join_probe_child,
+                hash_join_probe_data_types,
+                hash_join_probe_side_key_idxs,
+                build_side,
+                hash_join_build_data_types,
+                full_data_types,
+                hash_map,
+                next_build_row_with_same_key,
+            );
+
+            if let Some(cond) = self.condition.as_ref() {
+                let stream = match self.join_type {
+                    JoinType::Inner => {
+                        HashJoinExecutor::do_inner_join_with_non_equi_condition(params, cond)
+                    }
+                    JoinType::LeftOuter => {
+                        HashJoinExecutor::do_left_outer_join_with_non_equi_condition(params, cond)
+                    }
+                    JoinType::LeftSemi => {
+                        HashJoinExecutor::do_left_semi_join_with_non_equi_condition(params, cond)
+                    }
+                    JoinType::LeftAnti => {
+                        HashJoinExecutor::do_left_anti_join_with_non_equi_condition(params, cond)
+                    }
+                    JoinType::RightOuter
+                    | JoinType::RightSemi
+                    | JoinType::RightAnti
+                    | JoinType::FullOuter => unimplemented!(),
```

Comment:
Have these been prohibited from frontend?

Will it be supported in the future? Or never going to support it?

Reply:
Yes, this kind of join type had already prohibited from frontend. For `Right` join, I think a proper way is to convert it to the `Left` join by optimizer. For `FullOuter` join type , lookup join can not provide this kind of semantics. So never going to support it.